### PR TITLE
Updates to style and layout

### LINKED ISSUE #48: Style and Layout objectives for alpha release
There are at least a couple of style issues that should be resolved before an alpha release. I don't think the bar needs to be very high for style before the alpha release, but there's a few items that should be addressed.

* Fix the collision of Execute and Save buttons with argument inputs when models have a long list of inputs. (move the save buttons to the left sidebar under the "Setup" heading.)
* Add a visual cue to indicate the model is running for the second(s) after the Execute button is clicked but before the logfile is displayed
* Add a splash screen on startup so something can be displayed before the Flask app is online (1-2 seconds)

### LINKED ISSUE #49: Resources Tab links to community forum are broken
For most or all models, the "tag" for a model that is used to construct a URL is `undefined`. So URLs end up as:

https://community.naturalcapitalproject.org/tags/undefined

### PR DESCRIPTION
**Pull request by davemfish**
This PR fixes natcap/invest-workbench#48, fixes natcap/invest-workbench#49 .

In addition, it adds some new parsing and styling of invest logs in the `LogTab`. It improves the visual heirarchy of the logs by applying classes to text from the invest model's logger (as opposed to other loggers: taskgraph, pygeoprocessing, utils, etc), and to python exceptions.

This PR also adds a new `Portal` component, to allow more flexible placement of components outside the strict DOM heirarchy of their parent. https://reactjs.org/docs/portals.html

### CODE REVIEW

**Review by emlys — COMMENTED** [review]
Looking good! I like the addition of the spinning icon while the model is running. Just had a few comments, and I got the following warnings and error in the console logs.

When I click on the previous run (recreation) I get an error that might be related to the prop types changes in this PR:

> /Users/emily/invest-workbench-dave/node_modules/prop-types/checkPropTypes.js:20 Warning: Failed prop type: The prop `modelHumanName` is marked as required in `InvestJob`, but its value is `undefined`.
>     in InvestJob (created by App)
>     in App
> 

These warnings show up immediately upon starting the app and probably aren't related to this PR:
* `electron/js2c/renderer_init.js:15 (electron) Security Warning: webFrame.executeJavaScript was called without worldSafeExecuteJavaScript enabled. This is considered unsafe. worldSafeExecuteJavaScript will be enabled by default in Electron 12.`

* `electron/js2c/renderer_init.js:111 Electron Security Warning (Insecure Content-Security-Policy) This renderer process has either no Content Security Policy set or a policy with "unsafe-eval" enabled. This exposes users of this app to unnecessary security risks. For more information and help, consult https://electronjs.org/docs/tutorial/security. This warning will not show up once the app is packaged.`

* `(anonymous) @ electron/js2c/renderer_init.js:111
DevTools failed to load SourceMap: Could not load content for file:///Users/emily/invest-workbench-dave/node_modules/fecha/lib/fecha.umd.js.map: `

File: src/InvestJob.jsx [hunk 1]
```diff
@@ -217,11 +226,11 @@ export default class InvestJob extends React.Component {
         // TODO: handle case when job.logfile is still undefined?
         // Could be if some stdout is emitted before a logfile exists.
         logger.debug(`invest logging to: ${job.logfile}`);
-        job.status = 'running';
+        // job.status = 'running';
         this.setState(
           {
             logfile: job.logfile,
-            jobStatus: job.status,
+            // jobStatus: job.status,
```
[the remark this answers is not in the document]
Comment (davemfish) [reply src/InvestJob.jsx:233]: removed!

File: src/app.jsx [hunk 2]
```diff
@@ -99,18 +99,20 @@ export default class App extends React.Component {
...
    *
-   * @param  {string} modelRunName - invest model name as appears in `invest list`
+   * @param  {string} modelRunName - invest model name to be passed to `invest run`
+   * @param  {string} modelHumanName - the colloquial name of the invest model
    * @param  {object} argsValues - an invest "args dictionary" with initial values
    * @param  {string} logfile - path to an existing invest logfile
    * @param  {string} jobStatus - indicates how the job exited, if it's a recent job.
    */
-  openInvestModel(modelRunName, argsValues, logfile, jobStatus) {
+  openInvestModel(modelRunName, modelHumanName, argsValues, logfile, jobStatus) {
```
[the remark this answers is not in the document]
Comment (davemfish) [reply src/app.jsx:108]: Good question, I grappled with this a little bit. 

The two names are stored in `natcap.invest.cli` and returned if you run `invest list`. `modelRunName` can be passed to `invest run` and `modelHumanName` are the names appearing on the HomeTab buttons and on the Tab labels when a model is opened. 

Models also have `ARGS_SPEC.model_name`, which is not always identical to `modelHumanName` from `invest list`. Also, we need those names for the HomeTab buttons _before_ we fetch the model spec.

File: src/components/LogTab/index.jsx [hunk 3]
```diff
@@ -32,8 +36,11 @@ class LogDisplay extends React.Component {
 
   render() {
     return (
-      <Col ref={this.content} style={logStyle}>
-        {this.props.logdata}
+      <Col id="log-display" ref={this.content}>
+        <div
+          id="log-text"
+          dangerouslySetInnerHTML={{ __html: this.props.logdata }}
```
[the remark this answers is not in the document]
Comment (davemfish) [reply src/components/LogTab/index.jsx:42]: Good idea. I added a comment. Also, I'm open to ideas about how to do this in a better/safer way in the future.

**Comment by davemfish** [comment]
> When I click on the previous run (recreation) I get an error that might be related to the prop types changes in this PR:

This is probably due to some data cached locally in a `state_cache` directory from an older version of the workbench. On ubuntu that dir is in `~/.config/invest-workbench/state_cache`. It's safe to remove the whole folder anytime. Separately, I'm working on a refactor of how that persistent data is managed, and it sounds like it should include a check for compatibility before rendering those "recent job" buttons.

> These warnings show up immediately upon starting the app and probably aren't related to this PR:
> 
>     * `electron/js2c/renderer_init.js:15 (electron) Security Warning: webFrame.executeJavaScript was called without worldSafeExecuteJavaScript enabled. This is considered unsafe. worldSafeExecuteJavaScript will be enabled by default in Electron 12.`
> 
>     * `electron/js2c/renderer_init.js:111 Electron Security Warning (Insecure Content-Security-Policy) This renderer process has either no Content Security Policy set or a policy with "unsafe-eval" enabled. This exposes users of this app to unnecessary security risks. For more information and help, consult https://electronjs.org/docs/tutorial/security. This warning will not show up once the app is packaged.`

I made issue #60 for these. At first it seemed straightforward to select the "correct" security options, but then everything broke.

>     * `(anonymous) @ electron/js2c/renderer_init.js:111 DevTools failed to load SourceMap: Could not load content for file:///Users/emily/invest-workbench-dave/node_modules/fecha/lib/fecha.umd.js.map: `

I see this one too, but googling suggests it's not an issue with our codebase, but with fecha's. DevTools expects to find source maps, but couldn't find one for this library, I think.

**Comment by davemfish** [comment]
Thanks for the review @emlys , I just pushed some minor changes, so back to you!

**Comment by emlys** [comment]
> This is probably due to some data cached locally in a `state_cache` directory from an older version of the workbench. On ubuntu that dir is in `~/.config/invest-workbench/state_cache`. It's safe to remove the whole folder anytime. Separately, I'm working on a refactor of how that persistent data is managed, and it sounds like it should include a check for compatibility before rendering those "recent job" buttons.

Ah, great! Removing `~/Library/Application Support/invest-workbench/state_cache` did get rid of the error. Thanks for looking into those warnings too. Definitely nothing to do with this PR :)

**Review by emlys — APPROVED** [review]

**Comment by emlys** [comment]
Approved, but a little worried about that failing macOS test... looking into it now

**Pull request merged**